src/token/token.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* eslint-env browser */
+
+'use strict';
+
+import mix from '@ckeditor/ckeditor5-utils/src/mix';
+import ObservableMixin from '@ckeditor/ckeditor5-utils/src/observablemixin';
+
+const DEFAULT_OPTIONS = { refreshInterval: 3600000, autoRefresh: true };
+
+/**
+ * Class representing the token used for communication with CKEditor Cloud Services.
+ * Value of the token is retrieving from the specified URL and is refreshed every 1 hour by default.
+ *
+ * @mixes ObservableMixin
+ */
+class Token {
+	/**
+	 * Creates `Token` instance.
+	 *
+	 * @param {String} tokenUrl Endpoint address to download the token.
+	 * @param {Object} options
+	 * @param {String} [options.initTokenValue] Initial value of the token.
+	 * @param {Number} [options.refreshInterval=3600000] Delay between refreshes. Default 1 hour.
+	 * @param {Boolean} [options.autoRefresh=true] Specifies whether to start the refresh automatically.
+	 */
+	constructor( tokenUrl, options = DEFAULT_OPTIONS ) {
+		if ( !tokenUrl ) {
+			throw new Error( '`tokenUrl` must be provided' );
+		}
+
+		/**
+		 * Value of the token.
```

Comment:
I would add a note here that the value of the token is null until the token is requested if `initValue` is not provided and that you should call `init()` or `create()` to initialize token requesting.